pootle/apps/import_export/utils.py

```diff
@@ -49,8 +50,15 @@ def import_file(file, user=None):
                                 "Project/Language? (%s)", (file.name, e)))
 
     try:
+        tp = store.translation_project
+        allow_add_obsolete = (check_user_permission(user, 'administrate',
+                                                    tp.directory) and
+                              (tp.project.code == 'terminology' or
+                               tp.language.code == 'templates'))
```

Comment:
I am not sure how costly it is checking for a permission, but since `project` and `language` are most likely already cached it might make sense to check first for those and then for the permission to short circuit faster.